### PR TITLE
Adequacy patch CSR - revamp output variables [ANT-1932]

### PR DESCRIPTION
**Pull request by flomnes**
Backport of #2306

### CODE REVIEW

File: src/solver/optimisation/adequacy_patch_csr/set_variable_boundaries.cpp [hunk 1]
```diff
@@ -62,7 +62,7 @@ void HourlyCSRProblem::setBoundsOnENS()
                                           .ValeursHorairesDeDefaillancePositive[triggeredHour];
 
             AdresseDuResultat = &(problemeHebdo_->ResultatsHoraires[area]
-                                    .ValeursHorairesDeDefaillancePositive[triggeredHour]);
+                                    .ValeursHorairesDeDefaillancePositiveCSR[triggeredHour]);
```
Comment (guilpier-code) [inline src/solver/optimisation/adequacy_patch_csr/set_variable_boundaries.cpp]:
```c++
AdresseDuResultat = &(problemeHebdo_->ResultatsHoraires[area]
                                    .ValeursHorairesDeDefaillancePositiveCSR[triggeredHour]);
```
Don't know if it's expected, but on curent branch **develop**, we still have : 
```c++
AdresseDuResultat = &(problemeHebdo_->ResultatsHoraires[area]
                                    .ValeursHorairesDeDefaillancePositive[triggeredHour]);
```

File: src/solver/variable/economy/unsupliedEnergyCsr.h [hunk 2]
```diff
@@ -1,53 +1,53 @@
...
-namespace Antares::Solver::Variable::Economy
+namespace Antares
+{
+namespace Solver
+{
+namespace Variable
 {
-struct VCardSpilledEnergyAfterCSR
+namespace Economy
+{
```
Comment (guilpier-code) [inline src/solver/variable/economy/unsupliedEnergyCsr.h:32]: `namespace Antares::Solver::Variable::Economy` was replaced with brace-nested namespaces. 
I guess this **8.8.x** code is not supposed to be read again.

File: src/solver/variable/economy/unsupliedEnergyCsr.h [hunk 3]
```diff
@@ -1,53 +1,53 @@
...
     {
         return "MWh";
     }
+
     //! The short description of the variable
     static std::string Description()
     {
-        return "Spilled Energy After CSR Optimization (generation that cannot be satisfied) "
-               "after CSR optimization";
+        return "Unsuplied Energy after CSR (demand that cannot be satisfied)";
```
Comment (guilpier-code) [inline src/solver/variable/economy/unsupliedEnergyCsr.h:50]: `return "Unsuplied Energy after CSR (demand that cannot be satisfied)";` : 
In the description, we have "**after** CSR", is it intended ?

File: src/solver/variable/economy/unsupliedEnergyCsr.h (hunk 2 above)
Comment (flomnes) [reply src/solver/variable/economy/unsupliedEnergyCsr.h:32]: Yes

File: src/solver/variable/economy/unsupliedEnergyCsr.h (hunk 3 above)
Comment (flomnes) [reply src/solver/variable/economy/unsupliedEnergyCsr.h:50]: yes

**Pull request merged**